linters/importorder/main.go

```diff
@@ -0,0 +1,439 @@
+
+	ctx := build.Default
+	ctx.BuildTags = buildTags
+
+	conf := loader.Config{
+		Fset:  fs,
+		Build: &ctx,
+		// Since we are not concerned with the entire file, we should only parse the imports
+		ParserMode: parser.ImportsOnly,
+		// Continue even if type or IO errors are present
+		AllowErrors: true,
+	}
+
+	for _, importPath := range importPaths {
+		conf.ImportWithTests(importPath)
+	}
+
+	prog, err := conf.Load()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var groupedLintErrors []lintErrors
+	for _, pkg := range prog.InitialPackages() {
+		for _, file := range pkg.Files {
+			imports := imports(fs, file)
+			if len(imports) == 0 {
+				continue
+			}
+			fileLintErr := findErrors(imports, patterns, file)
+			if fileLintErr != nil {
+				groupedLintErrors = append(groupedLintErrors, fileLintErr)
+			}
+		}
+	}
+	return groupedLintErrors
+}
+
+// importDecl is the collection of importGroups contained in a single import block.
+type importDecl struct {
+	Groups []importGroup
+}
+
+// importGroup is a collection of imports
+type importGroup struct {
+	Imports []importSpec
+}
+
+// importSpec is a single import
+type importSpec struct {
+	Position token.Position
+	Line     int
+	Name     string
+	Path     string
+}
+
+// Imports returns the file imports grouped by paragraph.
+func imports(fset *token.FileSet, f *ast.File) []importDecl {
+	var importDecls []importDecl
+
+	for _, decl := range f.Decls {
+		genDecl, ok := decl.(*ast.GenDecl)
+		if !ok || genDecl.Tok != token.IMPORT {
+			break
+		}
+
+		var (
+			importDecl importDecl
+			group      importGroup
+		)
+
+		var lastLine int
+		for _, spec := range genDecl.Specs {
+			importSpec := spec.(*ast.ImportSpec)
+			pos := importSpec.Path.ValuePos
+			line := fset.Position(pos).Line
+			fileName := fset.Position(pos).Filename
+			if lastLine > 0 && pos > 0 && line-lastLine > 1 {
+				importDecl.Groups = append(importDecl.Groups, group)
+				group = importGroup{}
+			}
+			group.Imports = append(group.Imports, newImportSpec(importSpec, line, fileName))
+			lastLine = line
+		}
+		importDecl.Groups = append(importDecl.Groups, group)
+		importDecls = append(importDecls, importDecl)
+	}
+
+	return importDecls
+}
+
+func filterOutVendor(importPaths []string) []string {
+	filteredStrings := []string{}
+	for _, importPath := range importPaths {
+		if !strings.Contains(importPath, "/vendor/") {
+			filteredStrings = append(filteredStrings, importPath)
+		}
+	}
+	return filteredStrings
+}
+
+func newImportSpec(is *ast.ImportSpec, line int, filename string) importSpec {
+	var (
+		path string
+	)
+
+	pathLit := is.Path
+	if pathLit != nil {
+		path = pathLit.Value
+	}
+
+	return importSpec{
+		Name: filename,
+		Path: path,
+		Line: line,
+	}
+}
+
+func findErrors(importDecls []importDecl, patterns []string, file *ast.File) lintErrors {
+	var (
+		lintErrors lintErrors
+
+		// seenPatterns is used to keep track of patterns that have been seen already
+		seenPatterns = make([]string, 0)
```

Comment:
unless you're pre-declaring the size to be >0, why not just do `seenPatterns []string`?